js/app/location.js

```diff
@@ -0,0 +1,142 @@
+
+                return Promise.resolve({});
+            },
+            isInside: (renderer, mouse) => {
+                if (!mouse.isInitialize()) {
+                    return false;
+                }
+                var topLeftCorner = renderer.convertCoordonateToBackground(description.topLeftCorner);
+                var bottomRightCorner = renderer.convertCoordonateToBackground(description.bottomRightCorner);
+                return (
+                    mouse.getX() <= bottomRightCorner.x && mouse.getX() >= topLeftCorner.x &&
+                    mouse.getY() <= bottomRightCorner.y && mouse.getY() >= topLeftCorner.y
+                );
+            },
+        };
+
+        this.circle = {
+            dataCheck: () => {
+                CheckData.checkKeys(
+                    description,
+                    [
+                        'center',
+                        'center.x',
+                        'center.y',
+                        'radius',
+                    ],
+                    true,
+                    this.getName() + ' - Circle'
+                );
+            },
+            draw: (renderer, color) => {
+                var center = renderer.convertCoordonateToBackground(description.center);
+                var radius = renderer.convertValueToBackground(description.radius);
+                var canvasContext = renderer.getContext();
+
+                canvasContext.beginPath();
+                canvasContext.arc(center.x, center.y, radius, 0, 2 * Math.PI, false);
+                canvasContext.lineWidth = 3;
+                canvasContext.strokeStyle = color;
+                canvasContext.stroke();
+
+                return Promise.resolve({});
+            },
+            isInside: (renderer, mouse) => {
+                if (!mouse.isInitialize()) {
+                    return false;
+                }
+                var center = renderer.convertCoordonateToBackground(description.center);
+                var radius = renderer.convertValueToBackground(description.radius);
+                return Math.pow(mouse.getX() - center.x, 2) + Math.pow(mouse.getY() - center.y, 2)
+                    < Math.pow(radius, 2);
+            },
+        }
+
+        var handleUpdate = (renderer, mouse) => {
+            if (this[shape].isInside(renderer, mouse)) {
+                wasDrawn = true;
+                return this[shape].draw(renderer, colorInside);
+            } else if (!parent.isHidding()) {
+                wasDrawn = true;
+                return this[shape].draw(renderer, colorDefault);
+            } else {
+                var returnedData = {
+                    needRedrawScene: wasDrawn,
+                };
+                wasDrawn = false;
+                return Promise.resolve(returnedData);
+            }
+        }
+
+        this.render = (renderer, mouse) => {
+            return handleUpdate(renderer, mouse);
+        }
+
+        this.handleCursorMove = (renderer, mouse) => {
```

Comment:
Why can't you just write this.handleCursorMove = handleUpdate?

Reply:
in case i want to add more code later :)